zone/achievement_mutations.cpp

```
@@ -0,0 +1,98 @@
+	uint32_t achievement_id,
+	uint8_t component_type,
+	uint32_t component_id,
+	uint32_t value
+)
+{
+	const auto &manager = AchievementManager::Instance();
+	if (
+		!RuleB(Achievements, EnableAchievements) ||
+		!manager.IsLoaded() ||
+		!IsValidTarget(target_type, target_id) ||
+		component_type > 2 ||
```

Cppcheck (2.21.0)

[style] The function 'SetOpcode' is never used.

(unusedFunction)